File: <100/034/034.py
import math

factorial_of = [math.factorial(i) for i in range(10)]

def is_ok(n: int) -> bool:
    l = [int(c) for c in list(str(n))]
    l_factorial = [factorial_of[a] for a in l]
    s = sum(l_factorial)

    return s == n


def main() -> None:
    answer: int = 0
    upper_limit: int = 10**7
    
    for i in range(10, upper_limit):
        if is_ok(i):
            answer += i

    print(answer)
    return


if __name__ == '__main__':
    main()

File: <100/034/034.jl
function distribute_digits(n:: Int):: Array{Int}
  s = n |> string
  l:: Array{Int} = [parse(Int, c) for c ∈ s]
  return l
end

function is_ok(n:: Int):: Bool
  l:: Array{Int} = [parse(Int, c) for c ∈ (n |> string)]
  l_factorial:: Array{Int} = map(x -> factorial(x), l)
  s:: Int = l_factorial |> sum

  return s == n
end

function main():: Nothing
  answer:: Int = 0
  upper_limit:: Int = 10^7

  for i ∈ 10:upper_limit
    if i |> is_ok
      answer += i
    end
  end

  answer |> println
  return nothing
end

@time main()
